src/sidebar/permissions.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+var STORAGE_KEY = 'hypothesis.privacy';
+
+/**
+ * Object defining which principals can read, update and delete an annotation.
+ *
+ * This is the same as the `permissions` field retrieved on an annotation via
+ * the API.
+ *
+ * Principals are strings of the form `type:id` where `type` is `'acct'` (for a
+ * specific user) or `'group'` (for a group).
+ *
+ * @typedef Permissions
```

Comment:
What was the reason for using a `typedef` here rather than a class? I guess it's because these `Permissions` objects would not always be created by doing `new Permissions(...)`, sometimes preexisting ones would be extracted from `vm.annotation.permissions`?